app/Http/Controllers/Api/ReportbackController.php

```diff
@@ -23,15 +24,28 @@ public function __construct(ReportbackService $reportbackService)
     }
 
     /**
-     * Store a newly created resource in storage.
-     * @todo upsert reportbacks by default.
+     * Store a newly created resource in storage or
+     * update a reportback if it already exists.
+     *
      * @param  \Illuminate\Http\Request  $request
      * @return \Illuminate\Http\Response
      */
-    public function create(Request $request)
+    public function store(ReportbackRequest $request)
     {
-        $reportback = $this->reportbackService->create($request->all());
+        // @TODO - instead should probably just have a method that gets northstar_id by default from a drupal_id if that is the only thing provided and then use that to find the reportback.
```

Comment:
Yeah agreed! We have `normalize` and `resolve` methods on the Registrar in Northstar that handle this kind of "how do I find this thing from a bunch types of identifiers" problem and it seems to work pretty elegantly.